### PR TITLE
Default port for listen addr

### PR DESCRIPTION
**Pull request by dnephin**
Allow `--listen-addr a.x.y.z` to use a default port

Adds some unit tests for swarm options

### CODE REVIEW

**Comment by vieux** [comment]
LGTM

File: api/client/swarm/opts.go [hunk 1]
```diff
@@ -52,9 +72,19 @@ func (a *NodeAddrOption) Type() string {
 	return "node-addr"
 }
 
+// Value returns the value of this option as addr:port
+func (a *NodeAddrOption) Value() string {
+	return fmt.Sprintf("%s:%v", a.addr, a.port)
```
Comment (cpuguy83) [inline api/client/swarm/opts.go]: net.JoinHostPort?

File: api/client/swarm/opts.go [hunk 2]
```diff
@@ -25,25 +34,36 @@ var (
...
 
 // Set the value for this flag
 func (a *NodeAddrOption) Set(value string) error {
 	if !strings.Contains(value, ":") {
-		return fmt.Errorf("Invalid url, a host and port are required")
+		a.addr = value
+		return nil
 	}
 
 	parts := strings.Split(value, ":")
```
Comment (cpuguy83) [inline api/client/swarm/opts.go]: Seems like we should be using `net.SplitHostPort`, here.

Comment (justincormack) [inline api/client/swarm/opts.go]: Yes, got to be ipv6 compliant!

**Comment by tiborvass** [comment]
Addressed comments in #23703

File: api/client/swarm/opts.go (hunk 2 above)
Comment (dnephin) [inline api/client/swarm/opts.go]: Thanks, I thought this split felt wrong, but my quick search for such a function didn't turn up anything. Now I know where to find it.

**Pull request closed without merging**